### PR TITLE
Refactor E2E tests for Dashboard; add E2E tests for User and Organization settings pages

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- PARTIALLY implements https://github.com/enso-org/cloud-v2/issues/1232
  - Partially refactor E2E test actions into a state machine.
    - The main goal of this is to _disallow_ invalid actions - for example going from a page to itself, which will fail at runtime, or trying to create a new Data Link on a page where that button is not accessible.
    - An auxiliary goal is to have better namespacing of actions and better clarity:
      - Previously, everything was a locator at the top level of a single module. This makes it very difficult to comprehend what kinds of actions are available.
    - Note: There is also older `namespace`-based namespacing for the User and Organization settings pages, which were added before this refactor. They SHOULD be refactored to the new API, but I'm not sure whether it's worth spending the time right now.
  - Add E2E tests for every input on the "user" settings page and the "organization" settings page.
  - A skeletal E2E test for the Datalink modal has also been added - it does not actually test anything currently but should be sufficient for building upon.

### Important Notes
None

### Quality Assurance Recommendations
- Just look at a few tests and make sure they're actually checking things as they were previously.

### Code Review Recommendations
- The new API is in `e2e/actions/*`. Of note is the base class in `e2e/actions/BaseActions.ts`. It may also be worth checking one each of:
  - The subclasses (any `PascalCased` file)
  - Non-subclass groups of actions (any `camelCased` file ending in `Actions`)
  - And the one file that exports just a single action. (`openUserGroupAction`)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/authentication/cognito.mock.ts [hunk 1]
```diff
@@ -262,12 +262,15 @@ export class Cognito {
...
    * component. */
-  async changePassword(_oldPassword: string, _newPassword: string) {
+  async changePassword(oldPassword: string, newPassword: string) {
     const cognitoUserResult = await currentAuthenticatedUser()
     if (cognitoUserResult.ok) {
-      const result = await results.Result.wrapAsync(async () => {
-        // Ignored.
-      })
+      const result = await results.Result.wrapAsync(() =>
+        fetch('https://mock-cognito.com/change-password', {
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/authentication/cognito.mock.ts:269]: What's the mock-cognito.com ?

**Comment by PabloBuchu** [comment]
@MrFlashAccount @somebody1234 where are we with this PR? can we / do we want to merge this?

File: app/ide-desktop/lib/dashboard/src/authentication/cognito.mock.ts (hunk 1 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/cognito.mock.ts:269]: it's an arbitrary domain name that we use to tell the API shims defined in Playwright that we want something to happen on the backend

**Comment by MrFlashAccount** [comment]
@PabloBuchu it tests are passing, it's LGTM for me

**Review by MrFlashAccount — APPROVED** [review]

**Pull request merged**